src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,73 @@
-        return 0;
+        List<String[]> collect = numbers.stream()
+                .map(s -> s.split(","))
+                .collect(Collectors.toList());
+        return collect.stream()
+                .flatMap(Arrays::stream)
+                .mapToInt(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min()
+                .orElseThrow(() ->
+                        new RuntimeException("Can't get min value from list: "
+                                + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+
+        return IntStream.range(0, numbers.size())
+                .mapToDouble(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 != 0)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
     }
 
-    /**
-     * Given a List of `People` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new People(«Victor», 16, Sex.MAN),
-     * new People(«Helen», 42, Sex.WOMEN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<People> selectMenByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && p.getAge() <= toAge
+                        && p.getSex().equals(People.Sex.MAN))
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `People` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new People(«Victor», 16, Sex.MAN),
-     * new People(«Helen», 42, Sex.WOMEN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<People> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                 .filter(p -> p.getAge() >= fromAge
+                        && (p.getSex().equals(People.Sex.MAN)
+                         ? p.getAge() <= maleToAge :
+                        p.getAge() <= femaleToAge))
+                .collect(Collectors.toList());
```

Comment:
I think filter() looks too big, let`s create Predicate and use it in filter :

```suggestion
        Predicate<Person> predicatePerson = p -> p.getAge() >= fromAge
                        && (p.getSex().equals(People.Sex.MAN) ?  p.getAge() <= maleToAge 
                        : p.getAge() <= femaleToAge)
        return peopleList.stream()
                 .filter(predicatePerson)
                .collect(Collectors.toList());
```

Reply:
Created